seacrowd/sea_datasets/tgl_profanity/tgl_profanity.py

```diff
@@ -0,0 +1,112 @@
+    "val": "https://huggingface.co/datasets/mginoben/tagalog-profanity-dataset/resolve/main/val.csv",
+}
+
+
+class TagalogProfanityDataset(datasets.GeneratorBasedBuilder):
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "text"
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+    CLASS_LABELS = [1, 0]
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "label": datasets.Value("int64"),
+                }
+            )
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.text_features(label_names=self.CLASS_LABELS)
+        else:
+            raise ValueError(f"Invalid config name: {self.config.schema}")
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        data_files = dl_manager.download_and_extract(_URLS)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": data_files["train"]},
+            ),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": data_files["val"]}),
+        ]
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        """Yield examples as (key, example) tuples"""
+        with open(filepath, encoding="utf-8") as f:
+            csv_reader = csv.reader(f, delimiter=",")
+            next(csv_reader, None)  # skip the headers
+            for idx, row in enumerate(csv_reader):
+                text, label = row
+                if self.config.schema == "source":
+                    example = {"text": text, "label": int(label)}
+                elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+                    example = {"id": idx, "text": text, "label": label}
```

Comment:
from the way that you read the dataset and from these doccs:
1. [HF Docs on ClassLabel](https://huggingface.co/docs/datasets/v2.15.0/en/package_reference/main_classes#datasets.ClassLabel)
2. HF Datasets implementation (see these lines on [label definition](https://huggingface.co/datasets/glue/blob/main/glue.py#L456) and on [example generation](https://huggingface.co/datasets/glue/blob/main/glue.py#L597)) for `glue`

The label should be casted as `int` for SEACrowd Schema because the received `label` type is a string, and any field that uses type of ClassLabel are cast to `int`.

```suggestion
                    example = {"id": idx, "text": text, "label": int(label)}
```

Reply:
Ah I see, sorry for my confusion!